modin/experimental/engines/omnisci_on_ray/frame/partition_manager.py

```diff
@@ -47,17 +48,95 @@ def _compute_num_partitions(cls):
+
+        if len(unsupported_cols) > 0:
+            # Putting pandas frame into partitions instead of arrow table, because we know
+            # that all of operations with this frame will be default to pandas and don't want
+            # unnecessaries conversion pandas->arrow->pandas
+            return super().from_pandas(df, return_dims), unsupported_cols
+        else:
+            # Since we already have arrow table, putting it into partitions instead
+            # of pandas frame, to skip that phase when we will be putting our frame to OmniSci
+            return cls.from_arrow(at, return_dims, unsupported_cols)
+
+    @classmethod
+    def from_arrow(cls, at, return_dims=False, unsupported_cols=None):
         put_func = cls._partition_class.put_arrow
 
         parts = [[put_func(at)]]
+        if unsupported_cols is None:
+            _, unsupported_cols = cls._get_unsupported_cols(at)
 
         if not return_dims:
-            return np.array(parts)
+            result = np.array(parts)
         else:
             row_lengths = [at.num_rows]
             col_widths = [at.num_columns]
-            return np.array(parts), row_lengths, col_widths
+            result = np.array(parts), row_lengths, col_widths
+
+        return result, unsupported_cols
+
+    @classmethod
+    def _get_unsupported_cols(cls, obj):
+        """
+        Finds columns with unsupported dtypes by OmniSci, and returns list of it.
+
+        Parameters
+        ----------
+            obj : pandas.DataFrame or pyarrow.Table, object to inspect on
+                unsupported column types.
+
+        Returns
+        -------
+            Tuple of pyarrow.Table representation of `obj` (for future using) and
+            a list of unsupported columns.
+        """
+        if isinstance(obj, (pandas.Series, pandas.DataFrame)):
+
+            def fast_select_dtypes(df, dtype, nrows=None):
+                cols = [i for i, col in enumerate(df.dtypes.items()) if col[1] == dtype]
+                if nrows is None:
+                    nrows = len(df)
+                return df.iloc[nrows, cols]
+
+            # picking first rows from cols with `dtype="object"` to check its actual type,
+            # in case of homogen columns that saves us unnecessary convertion to arrow table
+            type_samples = fast_select_dtypes(obj, dtype="object", nrows=0)
+            unsupported_cols = [
+                name for name, col in type_samples.items() if not isinstance(col, str)
+            ]
+
+            if len(unsupported_cols) > 0:
+                return None, unsupported_cols
+
+            try:
+                obj = pyarrow.Table.from_pandas(obj)
+            except pyarrow.lib.ArrowTypeError as e:
+                regex = r"Conversion failed for column ([^\W]*)"
+                unsupported_cols = []
+                for msg in e.args:
+                    match = re.findall(regex, msg)
+                    unsupported_cols.extend(match)
```

Comment:
What would happen in case of a mismatched message?

Reply:
Yeah, in that case, we'll wrongly assume that there is no unsupported columns, because `len(unsupported_columns) == 0`, even though we've got `ArrowTypeError`. 

We can simply add a check after that for loop if `len(unsupported_columns) == 0` and if it is, then do `unsupported_columns = obj.columns`.

That will make warning about unsupported dtypes inaccurate, because there will be listed all columns of the frame, but if we can't get names of bad columns from arrow error, then we can't get it from anywhere.

Reply:
done that in the last commit